src/canopy/llm/openai.py

```diff
@@ -206,3 +207,135 @@ async def agenerate_queries(self,
                                 model_params: Optional[dict] = None,
                                 ) -> List[Query]:
         raise NotImplementedError()
+
+
+class AzureOpenAILLM(OpenAILLM):
+    """
+    Azure OpenAI LLM wrapper built on top of the OpenAI Python client.
+
+    Note: OpenAI requires a valid API key to use this class.
+          You can set the "AZURE_OPENAI_KEY" environment variable to your API key.
+          Or you can directly set it as follows:
+          >>> import openai
+          >>> openai.api_key = "YOUR_API_KEY"
+    """
+    def __init__(self,
+                 model_name: str = "gpt-3.5-turbo",
```

Comment:
Makes me wonder if I should change the implementation of `AzureOpenAILLM` to have different init arguments from the default `OpenAILLM` class? I added the `api_version` argument, does it make sense to rename `model_name` to `deployment_name`?

Reply:
@MichaelAnckaert `model_name` is part of the signature of all `BaseLLM` inheritors, so that is the only parameter that has to still exist in order to comply with `BaseLLM` interface.  
You can add additional init arguments or even change\remove ones that come from `OpenAILLM`, as long as you maintain compatibility with  `BaseLLM`.